sections/header.liquid

```diff
@@ -26,14 +26,17 @@
 <noscript>{{ 'component-cart-notification.css' | asset_url | stylesheet_tag }}</noscript>
 <noscript>{{ 'component-cart-items.css' | asset_url | stylesheet_tag }}</noscript>
 
+{%- assign logo_height = section.settings.logo_width | divided_by: section.settings.logo.aspect_ratio -%}
+
 <style>
   header-drawer {
     justify-self: start;
     margin-left: -1.2rem;
   }
 
   .header__heading-logo {
-    max-width: {{ section.settings.logo_width }}px;
+    height: {{ logo_height }}px;
+    width: {{ section.settings.logo_width }}px;
```

Comment:
I think the tricky part of making this `width` and not `max-width` is because of this:https://screenshot.click/24-05-4ysyc-w33ie.mp4

You can see in the video where on mobile it was being optimized before (beginning of the video) but now we have a way larger size.

We actually also have a PR to update language: https://github.com/Shopify/dawn/pull/1901

But by introducing this change, if the merchant selects 250, theyd end up with a really big logo on mobile

https://screenshot.click/24-07-sv528-5m2pi.png

cc @LucasLacerdaUX @Oliviammarcello

Reply:
I just tested making a change to the logo width in the theme settings and 250px looks equally bad on desktop. I don't think we should override what a merchant puts in the settings though. I might be misunderstanding your point so please let me know.

FYI, I put in a square logo. So by default, it's taller than the other menu elements. I didn't change the setting for logo width. So there is layout shift by default at the moment. I have a feeling that's a large use case so we need a way to prevent layout shift for taller logos.